mlkem/native/aarch64/ntt_clean.S

```diff
@@ -204,6 +214,20 @@ _ntt_asm_clean:
         load_roots_123
 
         .p2align 2
+
+        // Bounds reasoning:
+        // - There are 7 layers
+        // - When passing from layer N to layer N+1, each layer-N value
+        //   is modified through the addition/subtraction of a Montgomery
+        //   product of a twiddle of absolute value < q/2 and a layer-N value.
+        // - Recalling that |fqmul(a,t)| < q * (0.0254*C + 1/2) for
+        //   |a| < C*q and |t|<q/2, we see that the coefficients of layer N
+        //   (starting with layer 0 = input data) are bound by q * f^N(1), where
+        //   f(C) = 1/2 + 1.0508*C.
+        //   For N=7, we get the bound of f^7(1) * q < 18295.
```

Comment:
Isn't it easier to just show that for N=7, the upper bound is 7Q + 1664? That's a much simpler explanation, and still fits in 16 bits.

Reply:
Fair point. There are two parts here:
- Show that the arithmetic _within_ the NTT does not overflow.
- Exhibit a specific bound on the output of the NTT that is sufficient to show that _subsequent_ arithmetic does not overflow.

It seems to me that (a) basemul implementations currently need _no_ bound, (b) keygen merely needs `NTT_BOUND + KYBER_Q < INT16_MAX` to handle the addition https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/mlkem/indcpa.c#L292. That would actually be supported by a 8*Q bound. Still, I don't think it hurts to document a tighter bound in case changes in the subsequent arithmetic needs evaluation.

Reply:
You're right... if the inputs are bounded by +/- Q, then you must be within 8Q after 7 layers. I appreciate that your bound is tighter, but 8Q < 32767 is all we need, right?

Reply:
That's what my proofs in SPARK and C show anyway...

Reply:
No, it's note quite enough, because there's further arithmetic following the NTT: Specifically, there's https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/mlkem/indcpa.c#L292 adding a value up to size Q. So there one needs 9Q < INT16_MAX.

Reply:
Ah yes.